plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsBlobContainer.java

```diff
@@ -230,14 +243,26 @@ private void writeToPath(
         EnumSet<CreateFlag> createFlags
     ) throws IOException {
         final byte[] buffer = new byte[blobSize < bufferSize ? Math.toIntExact(blobSize) : bufferSize];
-        try (FSDataOutputStream stream = fileContext.create(blobPath, createFlags, CreateOpts.bufferSize(buffer.length))) {
+        Options.CreateOpts[] createOpts = getCreationOptions(CreateOpts.bufferSize(buffer.length));
+        try (FSDataOutputStream stream = fileContext.create(blobPath, createFlags, createOpts)) {
             int bytesRead;
             while ((bytesRead = inputStream.read(buffer)) != -1) {
                 stream.write(buffer, 0, bytesRead);
             }
         }
     }
 
+    private Options.CreateOpts[] getCreationOptions(Options.CreateOpts... baseOpts) {
+        if (replicationFactor == null) {
+            return baseOpts;
+        }
+
+        Options.CreateOpts[] newOpts = new Options.CreateOpts[baseOpts.length + 1];
```

Comment:
Nit: It might be nice to have a pre-assembled array of CreateOpts that is reused all the time and only to do the copy dance if both that array and the base opts array have properties? Right now we do it all the time if the replication factor is set, even if we don't have to. I imagine it's not a ton of overhead, but it feels clunky?